base/operators.jl

```diff
@@ -52,17 +52,18 @@ Generic equality operator. Falls back to [`===`](@ref).
 Should be implemented for all types with a notion of equality, based on the abstract value
 that an instance represents. For example, all numeric types are compared by numeric value,
 ignoring type. Strings are compared as sequences of characters, ignoring encoding.
-For collections, `==` is generally called recursively on all contents,
-though other properties (like the shape for arrays) may also be taken into account.
+Collections of the same type generally iterate through both operands and compare each pair
+of values, returning true if all such pairs are `==`, though other properties may also be
```

Comment:
```suggestion
Collections of the same type generally compare their key sets, and if those are `==`, then compare the values 
for each of those keys, returning true if all such pairs are `==`. Other properties are typically not taken into account (such as the exact type)
```

Reply:
For collections alone, this reads much better, yes.